### ISSUE TITLE
Improve Tailwind CSS implementation docs

### ISSUE THREAD

**Issue by danspratling** [gatsbyjs/gatsby#23952]
## Summary

Docs do not show how to use @tailwind directives
Docs do not show how to purge css (which is crucial for performance)

### Motivation

All creators looking to use Tailwind CSS in future projects will benefit from this change

### Draft the doc

N/A - doc already exists
https://www.gatsbyjs.org/docs/tailwind-css/#installing-and-configuring-tailwind

### Open a pull request

- [X] Open a pull request with your work

**Comment by jlarmstrongiv** [comment]
I remember I researched for a while trying to make a nice postcss.config.js.  I’m curious to see what your recommended config will be 😄 here’s the one I settled on:

```js
module.exports = {
  plugins: [
    require('postcss-import'),
    require('tailwindcss')('./tailwindcss.config.js'),
    // https://tailwindcss.com/docs/using-with-preprocessors#future-css-features
    require('postcss-preset-env')({
      stage: 1,
      features: {
        // https://github.com/tailwindcss/tailwindcss/issues/1190#issuecomment-546621554
        'focus-within-pseudo-class': false,
      },
      autoprefixer: process.env.NODE_ENV === 'production',
    }),

    // https://flaviocopes.com/tailwind-setup/
    process.env.NODE_ENV === 'production'
      ? require('cssnano')({ preset: 'default' })
      : null,
    process.env.NODE_ENV === 'production'
      ? require('@fullhuman/postcss-purgecss')({
          // Specify the paths to all of the template files in your project
          content: [
            './src/**/*.js',
            './src/**/*.jsx',
            './src/**/*.ts',
            './src/**/*.tsx',
          ],

          // Include any special characters you're using in this regular expression
          defaultExtractor: (content) =>
            content.match(/[\w-/.:]+(?<!:)/g) || [],
        })
      : null,
  ],
};
```

**Comment by danspratling** [comment]
The edits I've made actually don't change the suggested method for creating css/scss/css-in-js setups. They add more info to make sure you're not going to run into any roadblocks later.

If you feel the docs are missing a good config setup, it might be worth adding additional changes for review.

I used sass so I didn't need to make a postcss config, so this definitely doesn't apply to all use cases.

**Comment by danspratling** [comment]
I'm using scss so my entire config is this, with purge being handled directly within tailwind. I think personalisations should be handled by starters, rather than by the setup docs.

```js
{
  resolve: `gatsby-plugin-sass`,
  options: {
    postCssPlugins: [
      require('tailwindcss'),
      require('./tailwind.config.js'),
    ],
  },
},
```

I don't think any of the stuff which was already there needed much change, so I haven't updated it, but i was more concerned with ensuring the docs were fully fleshed out.

**Comment by gillkyle** [comment]
Closed by #23944
